### PR TITLE
test(gnoclient): Fix a potential bug in the integration test

### PR DESCRIPTION
**Pull request by linhpn99**
Screenshot from 2024-07-09 21-48-59

I'm not sure about the root cause, but it seems like the test node isn't ready to handle the request call, so it returns an error. Using the result immediately without asserting that there's no error will cause this section to panic, leading to a CI that fails randomly

You can see https://github.com/gnolang/gno/actions/runs/9855164413/job/27209502761?pr=2518

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Review by moul — APPROVED** [review]

File: gno.land/pkg/gnoclient/integration_test.go [hunk 1]
```diff
@@ -53,11 +53,11 @@ func TestCallSingle_Integration(t *testing.T) {
 
 	// Execute call
 	res, err := client.Call(baseCfg, msg)
+	assert.NoError(t, err)
```
Comment (thehowl) [inline gno.land/pkg/gnoclient/integration_test.go:56]:
```suggestion
	require.NoError(t, err)
```

File: gno.land/pkg/gnoclient/integration_test.go [hunk 2]
```diff
@@ -107,9 +107,9 @@ func TestCallMultiple_Integration(t *testing.T) {
 
 	// Execute call
 	res, err := client.Call(baseCfg, msg1, msg2)
+	assert.NoError(t, err)
```
Comment (thehowl) [inline gno.land/pkg/gnoclient/integration_test.go:110]:
```suggestion
	require.NoError(t, err)
```

File: gno.land/pkg/gnoclient/integration_test.go [hunk 3]
```diff
@@ -149,12 +149,12 @@ func TestSendSingle_Integration(t *testing.T) {
 
 	// Execute send
 	res, err := client.Send(baseCfg, msg)
-	assert.Nil(t, err)
+	assert.NoError(t, err)
```
Comment (thehowl) [inline gno.land/pkg/gnoclient/integration_test.go:152]:
```suggestion
	require.NoError(t, err)
```

File: gno.land/pkg/gnoclient/integration_test.go [hunk 4]
```diff
@@ -149,12 +149,12 @@ func TestSendSingle_Integration(t *testing.T) {
...
 	assert.Equal(t, "", string(res.DeliverTx.Data))
 
 	// Get the new account balance
 	account, _, err := client.QueryAccount(toAddress)
-	assert.Nil(t, err)
+	assert.NoError(t, err)
```
Comment (thehowl) [inline gno.land/pkg/gnoclient/integration_test.go:157]:
```suggestion
	require.NoError(t, err)
```

**Review by thehowl — COMMENTED** [review]
Using `assert` will fail the test, but still eventually panic, as asserts don't stop the execution of the test; `require` does.

**Pull request merged**